lib/Runtime/Base/jitprofiling.cpp

```diff
@@ -6,6 +6,12 @@
 
 #ifdef VTUNE_PROFILING
 
+// For Linux getenv is been defined as PAL_getenv in pal/inc/pal.h
+// We need undef here so that VTune can use it to get environment variable 
+#ifdef PLATFORM_UNIX
+#undef getenv    
```

Comment:
getenv is used at: https://github.com/Microsoft/ChakraCore/blob/master/lib/Runtime/Base/jitprofiling.cpp#L199_L201
VTune needs to retrieve environment variable via getenv in unix so that VTune JIT profiling can work, on Windows the environement variable is retrieved via GetEnvironmentVariableA, thus no problem on Windows. 
PAL_getenv fails because this file is used by both Windows and Linux, at the top of the file it needs include "RuntimeBasePch.h" so that it can be built correctly on Windows and Unix: https://github.com/Microsoft/ChakraCore/blob/master/lib/Runtime/Base/jitprofiling.cpp#L5, In Unix, "RuntimeBasePch.h" pulls in the include file of pal.h (pal/inc/pal.h), in that file pal.h: getenv is defined in PAL_getenv at https://github.com/Microsoft/ChakraCore/blob/master/pal/inc/pal.h#L6051, however PAL_getenv is empty when this jitprofiling.cpp is been built on Linux, thus causing VTune JIT Profiling failed. 
Another way to do it is to build pal cruntime earlier than building this component, since the implementation of PAL_getenv is at https://github.com/Microsoft/ChakraCore/blob/master/pal/src/cruntime/misc.cpp#L171_#L196. However, if we take this approach, not only we need make changes in the build procedure, we also need to make change in this jitprofiling.cpp file, change the original getenv to PAL_getenv.  For the purpose of easy maintenance in the future, I don't think it's a good idea to deviate too much from the original VTune file, especially making changes in the middle of the file. 
So compare to the 2nd approach, current approach is cleaner, less code change and less distruptive.